api/src/app/modules/project/project.service.ts

```diff
@@ -784,6 +784,105 @@ export class ProjectService extends DataService<Project, Repository<Project>, Pr
+  }
+
+
+  /**
+   * Retrieves the count of FOM projects grouped by district.
+   * Used by analytics dashboard module.
+   *
+   * @param startDate - The start of the date range (inclusive, YYYY-MM-DD)
+   * @param endDate - The end of the date range (inclusive, YYYY-MM-DD)
+   * @returns Promise resolving to an array of FOMCountByDistrictDto
+   */
+  async getProjectCountByDistrict(startDate: string, endDate: string): Promise<ProjectCountByDistrictResponse[]> {
+    return await this.repository
+      .createQueryBuilder('p')
+      .select('p.district_id', 'districtId')
+      .where('p.commenting_open_date >= :startDate', { startDate })
+      .andWhere('p.commenting_open_date <= :endDate', { endDate })
+      .andWhere('p.workflow_state_code != :workflowStateCode', { 
+        workflowStateCode: WorkflowStateEnum.INITIAL 
+      })
+      .innerJoin('p.district', 'd')
+      .addSelect('COUNT(p.project_id)', 'projectCount')
+      .addSelect('d.name', 'districtName')
+      .groupBy('p.district_id')
+      .addGroupBy('d.name')
+      .orderBy('"projectCount"', 'DESC')
+      .getRawMany();
+  }
+
+  /**
+   * Returns the total number of unique forest clients that have submitted FOMs.
+   * Excludes records where forest_client_number is null. Used by analytics dashboard module.
+   *
+   * @param startDate - The start of the date range (inclusive, YYYY-MM-DD)
+   * @param endDate - The end of the date range (inclusive, YYYY-MM-DD)
+   * @returns Promise resolving to the count of distinct forest client numbers
+   */
+  async getUniqueForestClientCount(startDate: string, endDate: string): Promise<number> {
+    return await this.repository
+      .createQueryBuilder()
+      .select('COUNT(DISTINCT forest_client_number)', 'count')
+      .where('forest_client_number IS NOT NULL')
+      .andWhere('commenting_open_date >= :startDate', { startDate })
+      .andWhere('commenting_open_date <= :endDate', { endDate })
+      .andWhere('workflow_state_code != :workflowStateCode', { 
+        workflowStateCode: WorkflowStateEnum.INITIAL 
+      })
+      .getRawOne()
+      .then(result => Number(result.count));
+  }
+
+  /**
+   * Retrieves the count of FOM projects submitted by each forest client.
+   * Used by analytics dashboard module.
+   *
+   * @param startDate - The start of the date range (inclusive, YYYY-MM-DD)
+   * @param endDate - The end of the date range (inclusive, YYYY-MM-DD)
+   * @returns Promise resolving to an array of ProjectCountByForestClientResponse
+   */
+  async getProjectCountByForestClient(
+    startDate: string,
+    endDate: string
+  ): Promise<ProjectCountByForestClientResponse[]> {
+    return await this.repository
+      .createQueryBuilder('p')
+      .innerJoin('p.forestClient', 'fc')
+      .select('p.forest_client_number', 'forestClientNumber')
+      .addSelect('fc.name', 'forestClientName')
+      .addSelect('COUNT(p.project_id)', 'projectCount')
+      .where('p.forest_client_number IS NOT NULL')
```

Comment:
I assume the forest client number will never be NULL in the project table?

Reply:
Yes, that's a required field.